src/Microsoft.Extensions.Configuration.DockerSecrets/DockerSecretsConfigurationProvider.cs

```diff
@@ -26,6 +26,13 @@ private static string NormalizeKey(string key)
             return key.Replace("__", ConfigurationPath.KeyDelimiter);
         }
 
+        private static string TrimNewLine(string value)
+        {
+            return value.EndsWith(Environment.NewLine)
+                ? value.Substring(0, value.Length - 1)
```

Comment:
To make this work cross plat, can you use value.Length - Environment.NewLine.Length so it works for both "\n" and "\r\n"?

Reply:
Done, @HaoK. Thank you for the feedback!